src/tests/ftest/util/systemctl_utils.py

```diff
@@ -0,0 +1,247 @@
+"""
+  (C) Copyright 2018-2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import getpass
+import os
+import tempfile
+
+from ClusterShell.NodeSet import NodeSet
+# pylint: disable=import-error,no-name-in-module
+from util.file_utils import create_directory, distribute_files
+from util.run_utils import command_as_user, run_remote
+
+
+class SystemctlFailure(Exception):
+    """Base exception for this module."""
+
+
+def get_service_status(logger, hosts, service):
+    """Get the status of the daos_server.service.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        hosts (NodeSet): hosts on which to get the service state
+        service (str): name of the service
+
+    Returns:
+        dict: a dictionary with the following keys:
+            - "status":       boolean set to True if status was obtained; False otherwise
+            - "stop":         NodeSet where to stop the daos_server.service
+            - "disable":      NodeSet where to disable the daos_server.service
+            - "reset-failed": NodeSet where to reset the daos_server.service
+
+    """
+    status = {
+        "status": True,
+        "stop": NodeSet(),
+        "disable": NodeSet(),
+        "reset-failed": NodeSet()}
+    status_states = {
+        "stop": ["active", "activating", "deactivating"],
+        "disable": ["active", "activating", "deactivating"],
+        "reset-failed": ["failed"]}
+    command = ["systemctl", "is-active", service]
+    result = run_remote(logger, hosts, " ".join(command))
+    for data in result.output:
+        if data.timeout:
+            status["status"] = False
+            status["stop"].add(data.hosts)
+            status["disable"].add(data.hosts)
+            status["reset-failed"].add(data.hosts)
+            logger.debug("  %s: TIMEOUT", data.hosts)
+            break
+        logger.debug("  %s: %s", data.hosts, "\n".join(data.stdout))
+        for key, state_list in status_states.items():
+            for line in data.stdout:
+                if line in state_list:
+                    status[key].add(data.hosts)
+                    break
+    return status
+
+
+def stop_service(logger, hosts, service):
+    """Stop any daos_server.service running on the hosts running servers.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        hosts (NodeSet): list of hosts on which to stop the service.
+        service (str): name of the service
+
+    Returns:
+        bool: True if the service was successfully stopped; False otherwise
+
+    """
+    result = {"status": True}
+    if hosts:
```

Comment:
IMO it's cleaner to do this instead of having to scroll past the code block just for one line of debug
```suggestion
    if not hosts:
        logger.debug(" Skipping stopping %s service - no hosts", service)
        return True
        
    result = {"status": True}
```

Reply:
Done.